tornado/ioloop.py

```diff
@@ -911,7 +914,7 @@ def _run(self) -> None:
     def _schedule_next(self) -> None:
         if self._running:
             self._update_next(self.io_loop.time())
-            self._timeout = self.io_loop.add_timeout(self._next_timeout, self._run)
+            self._timeout = self.io_loop.add_timeout(self._next_timeout, self._run)  # type: ignore
```

Comment:
We should think about why `# type: ignore` is needed here ... I ran into what I assume is a very similar case:

> error: Argument 2 to "call_later" of "IOLoop" has incompatible type "Callable[[str, str, str, str, str], Coroutine[Any, Any, None]]"; expected "Callable[..., None]"

`add_timeout()` and `call_later()` should accept Callables which return Coroutines, right @bdarnell ?
```

Reply:
That depends on whether `add_timeout` and friends will notice that a Coroutine has been returned and create a Task for it. We do this in `add_callback` but `add_timeout` and related methods just pass straight through to `asyncio`. I'm not sure what asyncio does here but I think it usually prefers task creation to be explicit. 

In any case, this new functionality needs a test, which will tell us whether `asyncio` is creating tasks here or not.